### PR TITLE
Fix status checks rendering

### PR DESCRIPTION
**Pull request by kabel**
* Properly converts GraphQL StatusContext and CheckRun types to PullRequestCheckStatus model
  * State is mapped from schema defined values to model values
  * Avatar URL is converted to enterprise-safe URL (was causing 404s)
* Await the mergable status from GraphQL is it was undefined in the REST list response
* Only set a required review as a failure when there are no other checks or there is review with requested changes

### CODE REVIEW

**Comment by kabel** [comment]
## Given

* An enterprise instance
* A repository clone with two open pull requests _that are able to be merged_

### Treeview render

_Before:_
image

_After:_
image

### Overview webview with the checks box expanded:

_Before:_
image

and after clicking the description treeview node to "refresh" the model/page:
image

_After:_
image

**Comment by Azcurgit** [comment]
I dont kno much but i need help with My Android is hacked

**Comment by Azcurgit** [comment]
My mail rodrigo.internetycorreo@gmail.com

**Comment by Azcurgit** [comment]
Ando My English is bad

**Comment by kabel** [comment]
In f394a8c753909849b766ecf3b7c9f46c5881453a, I included more fixes to better match the web version of the merge message container and tree view.

_Before_
image

_After_
image

The required reviews message is now pulled out of the checks list and is more accurately shown the right state:

image
image
image

**Comment by alexr00** [comment]
@kabel thank you for all the PRs. I've had a busy March so I haven't had a chance to give them a review yet. We're finishing up our endgame release process now through Monday, and I'm planning to catch up on all the PRs that need my attention starting tomorrow.

**Review by alexr00 — COMMENTED** [review]
👏 nice change, thank you! I've made a couple comments, which you can address in the next day or two if you want. Otherwise I plan to approve and merge after I release tomorrow.

File: src/github/githubRepository.ts [hunk 1]
```diff
@@ -1088,38 +1088,116 @@ export class GitHubRepository implements vscode.Disposable {
...
+				reviewRequirement = {
+					count: requiredApprovingReviews,
+					approvals: approvingReviews.map(review => review.author.login),
+					requestedChanges: requestedChanges.map(review => review.author.login),
+					state: state
+				};
+			}
+		}
 
+		return [checks, reviewRequirement];
```
[the remark this answers is not in the document]
Comment (kabel) [reply src/github/githubRepository.ts:1179]: Sure, I can widen the type of the returned _tuple_ to include null in the first element's type. However, I don't think that will have a matial change on the UI, as the merge component already took into account the lack of (empty array) any status checks.

[the remark this answers is not in the document]
Comment (kabel) [reply src/github/githubRepository.ts]: Example on PR of repo without status checks:

_Before_
image

_After_
image

**Review by alexr00 — APPROVED** [review]
Thanks for the changes!

**Review by aeschli — APPROVED** [review]

**Pull request merged**